cmake/opentelemetry-proto.cmake

```diff
@@ -1,27 +1,62 @@
-if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/third_party/opentelemetry-proto/.git)
-  set(PROTO_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/opentelemetry-proto")
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# The dependency on opentelemetry-proto can be provided different ways.
+# By order of decreasing priority, options are:
+#
+# 1 - Use a provided package
+#
+# This is useful to build opentelemetry-cpp as part of a super project.
+#
+# The super project provides the path to the opentelemetry-proto
+# source code using variable ${OTELCPP_PROTO_PATH}
+#
+# 2 - Search for a opentelemetry-proto git submodule
+#
+# When git submodule is used,
+# the opentelemetry-proto code is located in:
+# third_party/opentelemetry-proto
+#
+# 3 - Download opentelemetry-proto from github
+#
+# Code from the required version is used,
+# unless a specific release tag is provided
+# in variable ${opentelemetry-proto}
+#
+
+if(OTELCPP_PROTO_PATH)
+  message(STATUS "opentelemetry-proto dependency satisfied by: external path")
+  set(PROTO_PATH ${OTELCPP_PROTO_PATH})
```

Comment:
should this also validate the presence of common.proto , trace.proto and logs.proto, and error-out if missig?

Reply:
Good point, looking into this.

Reply:
Fixed:

Checked common.proto to detect, at configuration time (cmake), an invalid path, instead of having weird errors during the build.

Testing every proto file is probably overkill. If someone really modifies the content of opentelemetry-proto, things will break at compile time anyway, the same way it will break with a broken tag in git submodules, or a broken tag in github pull.